src/content/en/fundamentals/engage-and-retain/push-notifications/how-push-works.md

```diff
@@ -0,0 +1,160 @@
+valid.
+
+Each browser can use any push service they want, it's something developers have no control
+over. This isn't a problem because every push service expects the **same** API call. Meaning
+you don't have to care who the push service is, you just need to make sure that your API call
+is valid.
+
+To get the appropriate URL to trigger a push message (i.e. the URL for the push service) you
+just need to look at the `endpoint` value in a `PushSubscription`.
+
+Below is an example of the values you'll get from a **PushSubscription**:
+
+	{
+	  "endpoint": "https://random-push-service.com/some-kind-of-unique-id-1234/v2/",
+	  "keys": {
+	    "p256dh" :
+	"BNcRdreALRFXTkOOUHK1EtK2wtaz5Ry4YfYCA_0QTpQtUbVlUls0VJXg7A8u-Ts1XbjhazAkj7I99e8QcYP7DkM=",
+	    "auth"   : "tBHItJI5svbpez7KI4CCXg=="
+	  }
+	}
+
+The **endpoint** in this case is
+*https://random-push-service.com/some-kind-of-unique-id-1234/v2/*. The push service would be
+'random-push-service.com' and each endpoint is unique to a user, indicated with
+'some-kind-of-unique-id-1234'. As you start working with push you'll notice this pattern.
+
+We'll cover what the **keys** value is for later on.
+
+### What does the API look like?
+
+I mentioned that every web push service expects the same API call. That API is the [**Web Push
+Protocol**](https://tools.ietf.org/html/draft-ietf-webpush-protocol).
+
+It's an IETF standard that defines how you make an API call to a push service.
+
+The API call requires certain headers to be set and the data to be a stream of bytes. We'll
+look at libraries that can perform this API call for us as well as how to do it ourselves.
+
+### What can the API do?
+
+The API provides a way to send a message to a user, with / without data, and provide
+instructions of *how* to send the message.
+
+The data you send with a push message must be encrypted. The reason for this is that it
+prevents push services, who could be anyone, from being able to view the data sent with the
+push message. This is important given that it's the browser who decides which push service to
+use, which could open the door to browsers using a push service that isn't safe or secure.
+
+When you trigger a push message, the push service will receive the API call and queue the
+message. This message will remain queued until the users device comes online and the push
+service can deliver the messages. The instructions you can give to the push service define how
+the push message is queued.
+
+The instructions include details like:
+
+- The time-to-live for a push message. This defines how long a message should be queued before
+it's removed and not delivered.
+
+- Define the urgency of the message. This is useful in case the push service is preserving the
+users battery life by only delivering high priority messages.
+
+- Give a push message a "topic" name which will replace any pending message with this new message.
```

Comment:
You need parallel structure here and keep the items short. All three should be:

**Item**. A definition for the item.

Suggest: time-to-live, urgency, topic.

Also, Is 'topic' actually called 'topic'? I'm not fond of the term because (1) it sounds like a description, title, or something to display to the user, and (2) it comes out the client side as `tag`, making it a potential source of confusion.